// shortgo/dist/vgd.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __async = (__this, __arguments, generator) => {
  return new Promise((resolve, reject) => {
    var fulfilled = (value) => {
      try {
        step(generator.next(value));
      } catch (e) {
        reject(e);
      }
    };
    var rejected = (value) => {
      try {
        step(generator.throw(value));
      } catch (e) {
        reject(e);
      }
    };
    var step = (x) => x.done ? resolve(x.value) : Promise.resolve(x.value).then(fulfilled, rejected);
    step((generator = generator.apply(__this, __arguments)).next());
  });
};

// src/vgd.ts
var vgd_exports = {};
__export(vgd_exports, {
  vGdShorten: () => vGdShorten
});
module.exports = __toCommonJS(vgd_exports);
var import_axios = __toESM(require("axios"));
var import_zod = __toESM(require("zod"));
var shortenReturnSchema = import_zod.default.object({
  shorturl: import_zod.default.string().url()
});
var vGdShorten = (_0) => __async(void 0, [_0], function* ({ url, slug, advancedLog }) {
  try {
    if (slug && (slug.length < 5 || slug.length > 30 || !/^[a-zA-Z0-9_]*$/.test(slug))) {
      throw new Error(
        "Invalid slug. It must have between 5 and 30 alphanumeric characters or underscores"
      );
    }
    const shortenUrl = yield import_axios.default.get(
      `https://v.gd/create.php?format=json&url=${encodeURIComponent(url)}&${slug && `&shorturl=${slug}`}&${advancedLog && `&logstats=1`}`
    );
    return {
      ok: true,
      url: shortenUrl.data.shorturl,
      slug: shortenUrl.data.shorturl.split("/").pop()
    };
  } catch (error) {
    return {
      ok: false,
      error: "Invalid Shortener {v.gd}"
    };
  }
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  vGdShorten
});
